crates/bevy_utils/src/syncunsafecell.rs

```diff
@@ -76,11 +76,11 @@ impl<T: ?Sized> SyncUnsafeCell<T> {
     }
 
     #[inline]
-    /// Returns a `&SyncUnsafeCell<T>` from a `&mut T`.
-    pub fn from_mut(t: &mut T) -> &SyncUnsafeCell<T> {
+    /// Returns a `&mut SyncUnsafeCell<T>` from a `&mut T`.
+    pub fn from_mut(t: &mut T) -> &mut SyncUnsafeCell<T> {
         // SAFETY: `&mut` ensures unique access, and `UnsafeCell<T>` and `SyncUnsafeCell<T>`
         // have #[repr(transparent)]
-        unsafe { &*(t as *mut T as *const SyncUnsafeCell<T>) }
+        unsafe { &mut *(t as *mut T as *mut SyncUnsafeCell<T>) }
```

Comment:
Can we split this line so that we only keep the unsafe operation here? The chain of `as` conversions is pretty darn hard to scope when structured this way.

Reply:
I guess I could use `std::mem::transmute` instead of pointer casts, if that's clearer?

Reply:
Definitely not, and also much more prone to unsoundness. The only part here that is unsafe is the dereference cast at the end. The first `as` cast to `*mut T` is perfectly fine as is the second (which could probably be more clearly done via `ptr::cast<SyncUnsafeCell<T>>`) and can probably be moved outside of the `unsafe` block.